### PR TITLE
drop rxjava and completableFuture api

### PR DESCRIPTION
**Pull request by yiliuTo**
This pr is to drop rxjava and CompletableFuture support in spring cloud for azure messaging modules.

### CODE REVIEW

File: sdk/spring/azure-spring-servicebus/src/main/java/com/azure/spring/servicebus/core/queue/ServiceBusQueueTemplate.java [hunk 1]
```diff
@@ -60,13 +60,17 @@ protected void internalSubscribe(String name, Consumer<Message<?>> consumer, Cla
             this.checkpointConfig, payloadType, consumer, this.messageConverter) {
 
             @Override
-            protected String buildCheckpointFailMessage(Message<?> message) {
-                return String.format(MSG_FAIL_CHECKPOINT, message, name);
+            protected void buildCheckpointFailMessage(Message<?> message, Throwable t) {
```
Comment (saragluna) [inline sdk/spring/azure-spring-servicebus/src/main/java/com/azure/spring/servicebus/core/queue/ServiceBusQueueTemplate.java:63]: If this method is used to log information, should we change the method names accordingly?

File: sdk/spring/azure-spring-servicebus/src/main/java/com/azure/spring/servicebus/core/queue/ServiceBusQueueTemplate.java [hunk 2]
```diff
@@ -60,13 +60,17 @@ protected void internalSubscribe(String name, Consumer<Message<?>> consumer, Cla
...
+                if (LOGGER.isWarnEnabled()) {
+                    LOGGER.warn(String.format(MSG_FAIL_CHECKPOINT, message, name), t);
+                }
             }
 
             @Override
-            protected String buildCheckpointSuccessMessage(Message<?> message) {
-                return String.format(MSG_SUCCESS_CHECKPOINT, message, name, getCheckpointConfig().getCheckpointMode());
+            protected void buildCheckpointSuccessMessage(Message<?> message) {
```
Comment (saragluna) [inline sdk/spring/azure-spring-servicebus/src/main/java/com/azure/spring/servicebus/core/queue/ServiceBusQueueTemplate.java:70]: same here

File: sdk/spring/azure-spring-servicebus/src/main/java/com/azure/spring/servicebus/core/topic/ServiceBusTopicTemplate.java [hunk 3]
```diff
@@ -98,14 +98,18 @@ protected void internalSubscribe(String name,
         final DefaultServiceBusMessageProcessor messageProcessor = new DefaultServiceBusMessageProcessor(
             this.checkpointConfig, payloadType, consumer, this.messageConverter) {
             @Override
-            protected String buildCheckpointFailMessage(Message<?> message) {
-                return String.format(MSG_FAIL_CHECKPOINT, consumer, name, message);
+            protected void buildCheckpointFailMessage(Message<?> message, Throwable t) {
```
Comment (saragluna) [inline sdk/spring/azure-spring-servicebus/src/main/java/com/azure/spring/servicebus/core/topic/ServiceBusTopicTemplate.java:101]: same here

File: sdk/spring/azure-spring-servicebus/src/main/java/com/azure/spring/servicebus/core/topic/ServiceBusTopicTemplate.java [hunk 4]
```diff
@@ -98,14 +98,18 @@ protected void internalSubscribe(String name,
...
+                if (LOGGER.isWarnEnabled()) {
+                    LOGGER.warn(String.format(MSG_FAIL_CHECKPOINT, consumer, name, message), t);
+                }
             }
 
             @Override
-            protected String buildCheckpointSuccessMessage(Message<?> message) {
-                return String.format(MSG_SUCCESS_CHECKPOINT, consumer, name, message,
-                    getCheckpointConfig().getCheckpointMode());
+            protected void buildCheckpointSuccessMessage(Message<?> message) {
```
Comment (saragluna) [inline sdk/spring/azure-spring-servicebus/src/main/java/com/azure/spring/servicebus/core/topic/ServiceBusTopicTemplate.java:108]: same here

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/azure-spring-servicebus/src/main/java/com/azure/spring/servicebus/resourcemanager/ServiceBusQueueProvisioner.java [hunk 5]
```diff
@@ -1,7 +1,7 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-package com.azure.spring.servicebus.core;
+package com.azure.spring.servicebus.resourcemanager;
```
Comment (saragluna) [inline sdk/spring/azure-spring-servicebus/src/main/java/com/azure/spring/servicebus/resourcemanager/ServiceBusQueueProvisioner.java:4]: How about we name this package as xxx.provisioning, Just like org.springframework.cloud.stream.provisioning.ProvisioningProvider?

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/azure-spring-servicebus/src/main/java/com/azure/spring/servicebus/resourcemanager/ServiceBusQueueProvisioner.java (hunk 5 above)
Comment (yiliuTo) [reply sdk/spring/azure-spring-servicebus/src/main/java/com/azure/spring/servicebus/resourcemanager/ServiceBusQueueProvisioner.java:4]: Good idea!

**Review by stliu — APPROVED** [review]

**Review by saragluna — APPROVED** [review]
LGTM.

**Pull request merged**